### ISSUE TITLE
Should not require scheduler

### ISSUE THREAD

**Issue by siriuslee** [mosaicml/composer#428]
This may get to how we are defaulting a scheduler if none is provided, but specifying `max_duration='1ba'` should not require that the user specify a `Scheduler`. 

```
ValueError                                Traceback (most recent call last)
Input In [21], in <module>
----> 1 trainer = Trainer(model=model, 
      2                   train_dataloader=dataloader, 
      3                   eval_dataloader=None, 
      4                   max_duration="1ba", 
      5                   save_folder="checkpoints",
      6                   save_interval="1ep")
      7 trainer.fit()

File /usr/local/lib/python3.8/dist-packages/composer/trainer/trainer.py:297, in Trainer.__init__(self, model, train_dataloader, eval_dataloader, max_duration, algorithms, optimizers, schedulers, device, grad_accum, grad_clip_norm, validate_every_n_batches, validate_every_n_epochs, compute_training_metrics, precision, dist_timeout, ddp_sync_strategy, seed, deterministic_mode, log_destinations, callbacks, load_path, load_object_store, load_weights_only, load_strict, load_chunk_size, load_progress_bar, save_folder, save_interval, save_compression, profiler, train_subset_num_batches, eval_subset_num_batches, deepspeed_config)
    295 optimizer = ensure_tuple(optimizers)[0]
    296 if not max_duration.unit == TimeUnit.EPOCH:
--> 297     raise ValueError("If a scheduler is not provided, max duration must be in epochs")
    298 schedulers = CosineAnnealingLR(optimizer, T_max=max_duration.value)
    299 warnings.warn(f"No scheduler was specified. Defaulting to {repr(schedulers)}")

ValueError: If a scheduler is not provided, max duration must be in epochs
```

**Comment by ravi-mosaicml** [comment]
Curious what should the default behavior be if no scheduler is provided? Currently we are defaulting to `CosineAnnealingLR`, which requires max duration to be in epochs.

**Comment by ravi-mosaicml** [comment]
Fixed via #463, which changed the default behavior to be no scheduler.
